### PR TITLE
#142 Assign item to an employee

### PR DESCRIPTION
**Pull request by ammaratef45**
Started implementation of the assign function for #142
Added the model for the history table and a test for creating a new instance.
Left a puzzle to continue it.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#194](https://github.com/timelesslounge/timelessis/issues/194) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#194](https://github.com/timelesslounge/timelessis/issues/194) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/8b7e76e7-78d2-4874-8fc9-9d1f8f00ad1c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by vladarefiev** [comment]
@ammaratef45 Pls mention lint to issue

**Comment by ammaratef45** [comment]
@vladarefiev my fault, sorry

File: tests/test_models.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from timeless.roles.models import Role
 from timeless.schemetypes.models import SchemeType
 from timeless.items.models import Item
+from timeless.items.models import ItemHistory
```
Comment (vladarefiev) [inline tests/test_models.py]: @ammaratef45 pls do it in this way `from timeless.items.models import Item, ItemHistory `

File: tests/test_models.py [hunk 2]
```diff
@@ -228,3 +229,13 @@ def test_new_item():
         new_item.comment == comment and
         new_item.company_id == company_id
     )
+
+def test_new_item_history():
+    """ Test creation of new ItemHistory """
+    emp_id = 2
+    item_id = 1
+    item_history = ItemHistory(employee_id=emp_id, item_id=item_id)
+    assert (
```
Comment (vladarefiev) [inline tests/test_models.py]: @ammaratef45 let's do assert for every test:
`assert item_history.employee_id == emp_id`
`assert item_history.item_id == item_id`

File: timeless/items/models.py [hunk 3]
```diff
@@ -21,7 +18,36 @@ class Item(DB.Model):
     created_on = DB.Column(DB.DateTime, default=datetime.utcnow, nullable=False)
     updated_on = DB.Column(DB.DateTime, onupdate=datetime.utcnow)
     company = DB.relationship("Company", back_populates="items")
+    employee_id = DB.Column(DB.Integer, DB.ForeignKey("employees.id"))
+    empolyee = DB.relationship("Employee")
```
Comment (vladarefiev) [inline timeless/items/models.py]: @ammaratef45 Let's add reverse with `back_populates` or `backref `

File: timeless/items/models.py [hunk 4]
```diff
@@ -21,7 +18,36 @@ class Item(DB.Model):
...
+
+    def assignTo(self, employee):
```
Comment (vladarefiev) [inline timeless/items/models.py]: @ammaratef45 in python usually use snake case `def assign_to` or just `def assign` - good to me too

File: timeless/items/models.py [hunk 5]
```diff
@@ -21,7 +18,36 @@ class Item(DB.Model):
...
     def __repr__(self):
         """Return object information - String"""
         return "<Item %r>" % self.name
+
+class ItemHistory(DB.Model):
+    """Model for item assigning history
+    """
+    __tablename__ = "itemsHistory"
+
+
```
Comment (vladarefiev) [inline timeless/items/models.py]: @ammaratef45 excess line

File: timeless/items/models.py [hunk 6]
```diff
@@ -21,7 +18,36 @@ class Item(DB.Model):
...
+    id = DB.Column(DB.Integer, primary_key=True, autoincrement=True)
+    start_time = DB.Column(DB.DateTime, default=datetime.utcnow, nullable=False)
+    end_time = DB.Column(DB.DateTime)
+    employee_id = DB.Column(DB.Integer, DB.ForeignKey("employees.id"))
+    empolyee = DB.relationship("Employee")
```
Comment (vladarefiev) [inline timeless/items/models.py]: @ammaratef45 the same about reverse

File: timeless/items/models.py [hunk 7]
```diff
@@ -21,7 +18,36 @@ class Item(DB.Model):
...
+    item_id = DB.Column(DB.Integer, DB.ForeignKey("items.id"))
+    item = DB.relationship("Item")
+
+    def __repr__(self):
+        """Return object information - String"""
+        return "<Item %r>" % self.id
```
Comment (vladarefiev) [inline timeless/items/models.py]: @ammaratef45 let's do `return f"<Item {self.id}>"`

**Comment by vladarefiev** [comment]
@ammaratef45 Hey, I left a few comment. Check it out pls

**Comment by vladarefiev** [comment]
@ammaratef45 Hey, I left a few comment. Check it out pls

File: tests/test_models.py (hunk 2 above)
Comment (ammaratef45) [reply tests/test_models.py]: We should follow the one test, one assertion rule, right?

**Comment by ammaratef45** [comment]
@vladarefiev Can you see now?

**Comment by vladarefiev** [comment]
@rultor good to merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 35 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#194](https://github.com/timelesslounge/timelessis/issues/194) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@ammaratef45 According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your [message](https://github.com/timelesslounge/timelessis/pull/194#discussion_r255550195) by indicating an addressee in the beginning.

**Comment by ammaratef45** [comment]
@ypshenychka Done, sorry again 😅

**Comment by ypshenychka** [comment]
@ammaratef45 no problem, thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**